### PR TITLE
foundation for parsing anything from a discord message

### LINKED ISSUE #5: Strip mentions and server emotes when a discord message is tweeted

### PR DESCRIPTION
**Pull request by reallyrandom**
the cyclomatic complexity of the main function in bot.js is becoming unwieldy. I have not figured out the best way to refactor it to be more linear. I'd like to eliminate the paths by enumerating what should or more importantly should *not* be true for something to be tweeted or deleted. i can't see it off the top of my head in javascript though.

maybe this post has a decent solution: https://janalmazora.medium.com/how-to-prevent-using-if-else-statements-in-your-code-7e05e43afde

resolves #5

### CODE REVIEW

File: regex.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        console.log(`Found a match for ${regex} in ${message}`);
+        return true;
+    } else {
+        console.log(`Could not find a match for ${regex} in ${message}`);
+        return false;
+    }  
+}
+
+var StripRegexMatchFromMessage = exports.StripRegexMatchFromMessage = async function () {
+    return message.replace(regex)
```
Comment (william-sheldon) [inline regex.js:47]: missing a ;

File: regex.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    if (result) {
+        console.log(`Found a match for ${regex} in ${message}`);
+        return true;
+    } else {
+        console.log(`Could not find a match for ${regex} in ${message}`);
+        return false;
+    }  
+}
+
+var StripRegexMatchFromMessage = exports.StripRegexMatchFromMessage = async function () {
```
Comment (william-sheldon) [inline regex.js:46]: doesn't take any parameters

File: bot.js [hunk 3]
```diff
@@ -24,14 +24,21 @@ discordClient.on('messageReactionAdd', async (reaction) => {
...
-					console.log(`${reaction.message.author.username}'s message "${reaction.message.content}" will be tweeted.`);
-					const tweetUrl = await lib.PostTweet(records, twitterClient, reaction);
-					reaction.message.lineReplyNoMention(tweetUrl);
-					console.log(`Tweet sent, available at ${tweetUrl}.`);
-				} else {
 					console.log(`"${reaction.message.content}" has already been tweeted.`);
 					reaction.message.lineReplyNoMention("This message has already been twote.")
+					return
+				}
+				if (await regex.IdentifyRegexInMessage(regex.discord_emoji_regex, message)) {
```
Comment (william-sheldon) [inline bot.js:31]: i pretty much guess how to use global variables like this

**Comment by HUU** [comment]
I wrecked this entire PR by adding image support....sorry.

**Pull request closed without merging**